Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2480,6 +2482,15 @@ extension BrowserViewController: WKUIDelegate {
         openInHelper.open()
         decisionHandler(WKNavigationResponsePolicy.Cancel)
     }
+    
+    @available(iOS 9, *)
+    func webViewDidClose(webView: WKWebView) {
+        if let tab = tabManager[webView] {
+            dispatch_async(dispatch_get_main_queue()) {
```

Comment:
No need for this dispatch here; UI delegates are always executed on the main thread.